pkg/db/queries/envelopes_v2.sql.go

```diff
@@ -143,45 +143,86 @@ func (q *Queries) SelectGatewayEnvelopesByOriginators(ctx context.Context, arg S
 const selectGatewayEnvelopesByTopics = `-- name: SelectGatewayEnvelopesByTopics :many
 WITH cursors AS (
     SELECT x.node_id AS cursor_node_id, y.seq_id AS cursor_sequence_id
-    FROM unnest($3::INT[]) WITH ORDINALITY AS x(node_id, ord)
-             JOIN unnest($4::BIGINT[]) WITH ORDINALITY AS y(seq_id, ord)
+    FROM unnest($1::INT[])   WITH ORDINALITY AS x(node_id, ord)
+             JOIN unnest($2::BIGINT[]) WITH ORDINALITY AS y(seq_id, ord)
                   USING (ord)
-)
-SELECT v.originator_node_id,
-       v.originator_sequence_id,
-       v.gateway_time,
-       v.topic,
-       v.originator_envelope
-FROM gateway_envelopes_view v
-         LEFT JOIN cursors c
-                   ON v.originator_node_id = c.cursor_node_id
-WHERE v.topic = ANY($1::BYTEA[])
-  AND v.originator_sequence_id > COALESCE(c.cursor_sequence_id, 0)
-ORDER BY v.gateway_time, v.originator_node_id, v.originator_sequence_id
-LIMIT NULLIF($2::INT, 0)
+),
+     filtered AS (
+         -- A) topic + cursor
+         SELECT
+             m.originator_node_id,
+             m.originator_sequence_id,
+             m.gateway_time,
+             m.topic
+         FROM gateway_envelopes_meta AS m
+                  JOIN cursors AS c
+                       ON m.originator_node_id     = c.cursor_node_id
+                           AND m.originator_sequence_id > c.cursor_sequence_id
+         WHERE m.topic = ANY($4::BYTEA[])
+
+         UNION ALL
+
+         -- B) topic + no-cursor
+         SELECT
+             m.originator_node_id,
+             m.originator_sequence_id,
+             m.gateway_time,
+             m.topic
+         FROM gateway_envelopes_meta AS m
+         WHERE m.topic = ANY($4::BYTEA[])
+           AND m.originator_sequence_id > 0
```

Comment:
Is this pure defensiveness against negative sequence IDs, or does it impact query perf by encouraging the planner to use an index it might skip otherwise?